### ISSUE TITLE
Feature Request: Example Documentation

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#2782]
### Discussed in https://github.com/google/flax/discussions/2779

<div type='discussions-op-text'>

<sup>Originally posted by **shenoynikhil** January  8, 2023</sup>
In the [API Reference](https://flax.readthedocs.io/en/latest/api_reference/index.html), is it possible to have an example for each module. It's generally easy to understand the usage from an example. 

The request is inspired from pytorch (example shared below for `torch.nn.Conv2D` [Link](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html))
<img width="801" alt="Screenshot 2023-01-08 at 2 29 15 PM" src="https://user-images.githubusercontent.com/28651490/211222324-4d1486c4-71f7-43a4-a316-28eb782560a7.png">

I am also willing to help out to improve the documentation if I can get some support :).</div>

**Comment by IvyZX** [comment]
Most of the modules actually are used in [flax/examples](https://github.com/google/flax/tree/main/examples) or in some documentation page. It might be a good practice to at least put a link on the docstring so that users can see it from API Reference. 

For example, from what I've seen for now, `Dense` and `Conv` are used in [Getting Started](https://flax.readthedocs.io/en/latest/getting_started.html), and `BatchNorm` and `Dropout` have their own guides.

**Comment by shenoynikhil** [comment]
I'll be glad to help improve the documentation.

**Comment by cgarciae** [comment]
There are efforts on this direction (e.g. #2792) however it will take time to improve all docstrings so I set the tag for this issue to _eventual_.
